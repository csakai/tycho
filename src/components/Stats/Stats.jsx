import React from 'react';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';

export default class Stats extends React.Component {

    static propTypes = {
        description: PropTypes.string,
        velocity: PropTypes.string,
        magnitude: PropTypes.string,
        trueAnomaly: PropTypes.string,
        time: PropTypes.string
    }

    render() {
        return (
            <div className="stats">
                <div className="stats__content">
                    <ReactMarkdown source={this.props.description || ''} />
                </div>

                <div className="stats__footer">
                    <div className="stats__info">
                        <span>{this.props.pageText.stats.currentEarthTime}</span><br />
                        <span>{this.props.pageText.stats.velocityAtVector}</span><br />
                        <span>{this.props.pageText.stats.distanceToSun}</span><br />
                        <span>{this.props.pageText.stats.trueAnomaly}</span><br />
                    </div>
                    <div className="stats__info">
                        <span>{this.props.time}</span><br />
                        <span>
                            {this.props.velocity}&nbsp;
                            {this.props.pageText.abbreviations.kilometers}/
                            {this.props.pageText.abbreviations.seconds}
                        </span><br />
                        <span>{this.props.magnitude} {this.props.pageText.abbreviations.kilometers}</span><br />
                        <span>{this.props.trueAnomaly} {this.props.pageText.abbreviations.theta}</span>
                    </div>
                </div>
            </div>
        );
    }
}