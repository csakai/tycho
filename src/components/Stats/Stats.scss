@import 'globals.scss';

.stats {
    display: flex;
    flex-direction: column;
    width: 100%;

    &__info {
        flex: 1;
        line-height: $font-line-height;
    }

    &__footer {
        padding-top: $base-spacing;
        margin-top: $base-spacing;
        border-top: $dotted-border;
        display: flex;
    }
}
