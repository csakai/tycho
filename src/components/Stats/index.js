import './Stats.css';
export {default} from './Stats';