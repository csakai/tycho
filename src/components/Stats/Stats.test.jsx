import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Stats from './Stats';

describe('Stats', () => {
    const pageText = {
        stats: {},
        abbreviations: {}
    };

    it('should render the stats component with a blank description', () => {
        const component = shallow(<Stats
            pageText={pageText}
            description="Hello, world."
        />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render the stats component with the given description', () => {
        const component = shallow(<Stats pageText={pageText} />);
        expect(toJson(component)).toMatchSnapshot();
    });
});