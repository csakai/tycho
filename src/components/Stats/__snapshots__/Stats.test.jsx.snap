// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats should render the stats component with a blank description 1`] = `
<div
  className="stats"
>
  <Markdown
    text="Hello, world."
  />
  <div
    className="stats__footer"
  >
    <div
      className="stats__info"
    >
      <span />
      <br />
      <span />
      <br />
      <span />
      <br />
      <span />
      <br />
    </div>
    <div
      className="stats__info"
    >
      <span />
      <br />
      <span>
         
        /
      </span>
      <br />
      <span>
         
      </span>
      <br />
      <span>
         
      </span>
    </div>
  </div>
</div>
`;

exports[`Stats should render the stats component with the given description 1`] = `
<div
  className="stats"
>
  <Markdown
    text=""
  />
  <div
    className="stats__footer"
  >
    <div
      className="stats__info"
    >
      <span />
      <br />
      <span />
      <br />
      <span />
      <br />
      <span />
      <br />
    </div>
    <div
      className="stats__info"
    >
      <span />
      <br />
      <span>
         
        /
      </span>
      <br />
      <span>
         
      </span>
      <br />
      <span>
         
      </span>
    </div>
  </div>
</div>
`;
